### PR TITLE
File/Folder Reorg Changes

### PR DESCRIPTION
**Pull request by arvindcheru**
- File/Folder Reorg Changes Applied
- Install target rocm_path (with out prefix)
- Bin path updated (with out prefix)
- Data Path updated (with out prefix)

### CODE REVIEW

**Review by pfultz2 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -50,14 +50,47 @@ project ( MIOpen C CXX )
...
+
+  execute_process(COMMAND ${CMAKE_COMMAND} -E tar xzvf "${rocm_cmake_archive}"
+    WORKING_DIRECTORY ${PROJECT_EXTERN_DIR})
+  execute_process( COMMAND ${CMAKE_COMMAND} -DCMAKE_INSTALL_PREFIX=${PROJECT_EXTERN_DIR}/rocm-cmake .
+    WORKING_DIRECTORY ${PROJECT_EXTERN_DIR}/rocm-cmake-${rocm_cmake_tag} )
+  execute_process( COMMAND ${CMAKE_COMMAND} --build rocm-cmake-${rocm_cmake_tag} --target install
+    WORKING_DIRECTORY ${PROJECT_EXTERN_DIR})
+
+  find_package( ROCM 0.7.3 REQUIRED CONFIG PATHS ${PROJECT_EXTERN_DIR}/rocm-cmake )
+endif()
```
[the remark this answers is not in the document]
Comment (arvindcheru) [reply CMakeLists.txt:83]: Thank for the comments, I will remove this from here.

**Comment by junliume** [comment]
Ping @pfultz2  and @JehandadKhan  for review.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -46,6 +46,10 @@ configure_file("${PROJECT_SOURCE_DIR}/include/miopen/config.h.in" "${PROJECT_BIN
 # configure a header file to pass the CMake version settings to the source, and package the header files in the output archive
 configure_file( "${PROJECT_SOURCE_DIR}/include/miopen/version.h.in" "${PROJECT_BINARY_DIR}/include/miopen/version.h" )
 
+#Copy header to Build Dir
+configure_file( "${PROJECT_SOURCE_DIR}/include/miopen/miopen.h" "${PROJECT_BINARY_DIR}/include/miopen/miopen.h" )
```
Comment (pfultz2) [inline src/CMakeLists.txt:50]: Can a `if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)` be put around this? When we remove this option then this configure_file should be removed as well.

**Review by JehandadKhan — DISMISSED** [review]

File: src/CMakeLists.txt (hunk 2 above)
Comment (junliume) [reply src/CMakeLists.txt:52]: @arvindcheru to raise awareness of the above comment.

Comment (arvindcheru) [reply src/CMakeLists.txt:52]: Thanks for the comments from @pfultz2 , @junliume . Current implementation for supporting backward compatibility we have used the build directory in line 764 (below snippet)
=======
 if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
  rocm_wrap_header_dir(
    ${PROJECT_BINARY_DIR}/include/miopen
========================
So to make available all headers we were copying the missing file "miopen.h" in line 50 above.
So As per comment from Paul, we have changed this line under if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY).

**Review by junliume — APPROVED** [review]

**Comment by arvindcheru** [comment]
Had to update on top one fix for correcting include location - in the last version it was defaulted to /include/MIOpen/ (generated from Projectname), this is corrected in the latest version
733b91ed1a5403aefb7af392371124b4c503f406

Thank you for review approvals  @junliume @JehandadKhan 
Kindly take the last version for merge.

**Review by junliume — APPROVED** [review]

**Review by lawruble13 — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Comment by atamazov** [comment]
It breaks build on vanilla ROCm 5.0. Ticket to follow.

**Pull request merged**